### PR TITLE
Refactor bearer tokens & recovery codes

### PR DESCRIPTION
**Pull request by kiootic**
ref #73

### CODE REVIEW

File: docs/specs/user-model.md [hunk 1]
```diff
@@ -238,8 +238,8 @@ Out-of-band One-time-password authenticator is either primary or secondary.
 
 OOB-OTP authenticator is bound to a recipient address. The recipient can be an email address or phone number that can receive SMS messages.
 
-An OOB-OTP authenticator may be bound to a login ID identity. If the bounded
-identity is deleted, the authenticator would be deleted as well.
+An OOB-OTP authenticator may matches a login ID identity. When a login ID
+identity is deleted, any OOB-OTP matching authenticator would be deleted as well.
```
Comment (louischan-oursky) [inline docs/specs/user-model.md:242]: Suppose the user signed up with OAuth. Later they added OOB-OTP authenticator as 2 factor. And then they added the login ID identity matching the OOB-OTP authenticator. When the login ID identity is removed, the authenticator should be kept.

File: docs/specs/user-model.md [hunk 2]
```diff
@@ -261,18 +261,25 @@ authenticator:
...
+secondary authentication as long as it remains valid.
 
-The token is a cryptographically secure random string of 256 bits (32 bytes) in hex encoding.
+The token is a cryptographically secure random string with at least 256 bits.
 
-### Recovery Code Authenticator
+### Recovery Code
 
-Recovery code authenticator is secondary.
+Recovery codes are used to skip secondary authentication when a secondary
```
Comment (louischan-oursky) [inline docs/specs/user-model.md:276]: It is not accurate to say recovery code is used to skip secondary authentication.

File: docs/specs/user-model.md (hunk 1 above)
Comment (kiootic) [reply docs/specs/user-model.md:242]: If we don't delete it, when user removed a login ID identity, the 'verification status' would remain, so if the same identity is re-added, it is verified immediately.

Comment (louischan-oursky) [reply docs/specs/user-model.md:242]: The verification status is `matched && (created_at of authenticator > created_at of identity)`?

Comment (kiootic) [reply docs/specs/user-model.md:242]: In that case, when user attempt to verify the re-added identity, we'll remove the existing authenticator and add a new authenticator?

Comment (louischan-oursky) [reply docs/specs/user-model.md]: Or just update the created_at column (though it seems weird)

**Pull request merged**